entities/component.py

```diff
@@ -115,9 +117,11 @@ def __str__(self) -> str:
         ret = '(gate {}\n'.format(self.uuid) +\
             ' {}\n'.format(self.symbol_uuid) +\
             ' {} {} {} {}\n'.format(self.position, self.rotation, self.required, self.suffix)
+        pin_lines = []
         for pin in self.pins:
-            ret += ' {}\n'.format(pin)
-        ret += ')'
+            pin_lines.append(' {}'.format(pin))
+        ret += '\n'.join(sorted(pin_lines))
+        ret += '\n)'
```

Comment:
I guess this is required from LibrePCB that the pins are sorted?

Reply:
Yep, pin sorting is required for canonical output!